lib/vdsm/storage/lvm.py

```diff
@@ -657,10 +660,20 @@ def _reload_single_vg(self, vg_name):
             if error:
                 self._update_stale_vgs([vg_name])
 
+                if not out:
+                    # Reload a specific VG name and failing without
+                    # an output might be indicative of a real error.
+                    raise se.VolumeGroupDoesNotExist(vg_name, error=error)
```

Comment:
One small detail to make a more useful traceback. If we do:

    raise se.VolumeGroupDoesNotExist(vg_name, error=error).with_exception(error)

We will have single traceback for the entire call - from the caller frame to the frame calling the lvm command. In the current way we have 2 tracebacks - one from the call frame to this point, and another from this point to the call to lvm command.

Maybe we need a class method like LVMCommandError.from_error(), that will set the instance error attribute, and copy the error.__traceback__.

Reply:
Yes, keeping the traceback is always a good idea. We will benefit from it the next time it fails, and some user reports with logs.